### PR TITLE
Product Query: Add patterns

### LINKED ISSUE #7624: Product Query [3]: Implement first set of patterns specific to the Product Query block
A first set of patterns provided by design should be implemented as only available within the Product Query block. See https://github.com/WordPress/gutenberg/pull/44197 on how this is possible.

CC @vivialice to provide the patterns.

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR adds Product Query patterns based on Figma designs.

Fixes #7624

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Create a new page.
2. Add each pattern and make sure it displays correctly in the editor and frontend.

**1. Product gallery**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|All_Products_–_ratings-11|

**2. 4-Column Product Row**

| Figma | Pattern | 
|------ | ----- |
| 🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**3. 3-Column Product Row**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**4. Large Image Product Gallery**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**5. Minimal 5-Column Products Row**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**6. 1:1 Image 4-Column Products Row**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**7. Product list with full product description**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**8. Product list with 1:1 images**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

**9. Minimal product list**

| Figma | Pattern | 
|------ | ----- |
|🔎_Product_Query_Block_–_Figma|Products_–_ratings|

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Products block: Add patterns.

### CODE REVIEW

**Comment by nerrad** [comment]
For the rating block, is there a way to change the default rendering when there are no ratings to show 5 empty stars instead of blank? It would help with preserving the layout and eventually be a consistent way to link shoppers directly to the rating results for the product.

**Comment by danieldudzic** [comment]
> For the rating block, is there a way to change the default rendering when there are no ratings to show 5 empty stars instead of blank? It would help with preserving the layout and eventually be a consistent way to link shoppers directly to the rating results for the product.

This is a really good suggestion. I'll reach out to @kmanijak and see if he can prioritize this.

**Comment by kmanijak** [comment]
@nerrad , @danieldudzic I [created an issue](https://github.com/woocommerce/woocommerce-blocks/issues/7923), I'll kick it off tomorrow 👍

File: patterns/product-query-minimal-product-list.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+<div class="wp-block-columns"><!-- wp:column {"width":"33.33%"} -->
+<div class="wp-block-column" style="flex-basis:33.33%"><!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column {"width":"66.66%"} -->
+<div class="wp-block-column" style="flex-basis:66.66%"></div>
+<!-- /wp:column --></div>
+<!-- /wp:columns -->
```
Comment (Aljullu) [inline patterns/product-query-minimal-product-list.php]: These columns are empty, are they here on purpose or should we remove them?

imatge

File: patterns/product-query-product-list-with-1-1-images.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+<div class="wp-block-columns"><!-- wp:column {"width":"33.33%"} -->
+<div class="wp-block-column" style="flex-basis:33.33%"><!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column {"width":"66.66%"} -->
+<div class="wp-block-column" style="flex-basis:66.66%"></div>
+<!-- /wp:column --></div>
+<!-- /wp:columns -->
```
Comment (Aljullu) [inline patterns/product-query-product-list-with-1-1-images.php]: Same here.

**Review by Aljullu — APPROVED** [review]
I left a couple of inline comments, but overall looks good to me, so pre-approving.

Two more things which I don't consider blocking:

* I think we should review Product elements margins altogether. Ideally we should remove the margins from CSS and use global styles for that. It's tricky because we have other blocks that rely on some of these margins and it's an issue that was also present in the All Products block (https://github.com/woocommerce/woocommerce-blocks/issues/2565).
* In some occasions, there is [content shift when selecting inner blocks](https://user-images.githubusercontent.com/3616980/207310245-e4d0036d-4d7c-423d-9471-339a65aaf85b.webm). It might be related to the CSS margin issues we have, or it might be an issue in GB and not in our repo, I have no idea. But it might be worth creating an issue to investigate it.

**Pull request merged**